utils/common/src/main/java/brooklyn/util/crypto/AuthorizedKeysParser.java

```
@@ -0,0 +1,132 @@
+            // both ssh-rsa and ssh-dss begin with "AAAA" due to the length bytes
+            for (String part : keyLine.split(" ")) {
+                if (part.startsWith("AAAA")) {
+                    stream = new ByteArrayInputStream(BaseEncoding.base64().decode(part));
+                    break;
+                }
+            }
+            if (stream == null)
+                throw new IllegalArgumentException("Encoded public key should include phrase beginning AAAA.");
+
+            String type = readType(stream);
+            if (type.equals("ssh-rsa")) {
+                BigInteger e = readBigInt(stream);
+                BigInteger m = readBigInt(stream);
+                RSAPublicKeySpec spec = new RSAPublicKeySpec(m, e);
+                return KeyFactory.getInstance("RSA").generatePublic(spec);
+            } else if (type.equals("ssh-dss")) {
+                BigInteger p = readBigInt(stream);
+                BigInteger q = readBigInt(stream);
+                BigInteger g = readBigInt(stream);
+                BigInteger y = readBigInt(stream);
+                DSAPublicKeySpec spec = new DSAPublicKeySpec(y, p, q, g);
+                return KeyFactory.getInstance("DSA").generatePublic(spec);
+            } else {
+                throw new IllegalArgumentException("Unknown public key type " + type);
+            }
+            
+        } catch (Exception e) {
+            Exceptions.propagateIfFatal(e);
+            throw new IllegalArgumentException("Error parsing authorized_keys/SSH2 format public key: "+e);
+        }
+    }
+
+    private static int readInt(InputStream stream) throws IOException {
+        return ((stream.read() & 0xFF) << 24) | ((stream.read() & 0xFF) << 16)
+            | ((stream.read() & 0xFF) << 8) | (stream.read() & 0xFF);
+    }
+    
+    private static byte[] readBytesWithLength(InputStream stream) throws IOException {
+        int len = readInt(stream);
+        byte[] result = new byte[len];
```

Perhaps guard this - if `len` is big then there's something wrong (given the context of this method). Better to throw a graceful exception rather than cause an OutOfMemoryError.